src/chapter3/software-tooling.md

```diff
@@ -0,0 +1,91 @@
+# Software and Tooling
+
+Software and development tooling is handled a little differently on M3 than you might be used to. In particular, because M3 is a shared file system, you do not have access to `sudo`, and you cannot install software on the system manually. Instead, you will need to use the `module` command to load software and development tools.
+
+## Module
+
+The `module` command is used kind of as an alternative to package managers like `apt` or `yum`, except it is managed by the M3 team. It allows you to load software and development tools into your environment, and is used to load software on M3. To see a comprehensive list of commands go [here](./linux-cmds.md#m3-specific-commands).
+
+In general, however, you will only really need to use `module load` and `module unload`. These commands are used to load and unload software and development tools into your environment.
+
+For most of the more popular software packages, like gcc, there are multiple different versions available. You will need to specify which version you want to load based on your needs.
+
+## C / C++
```

Comment:
This is a bit of a pedantic thing but don't group C and C++ together. Because they are so different (as different as Python is to Java at this point) I don't want us to teach the misconception that C/C++ is a language. Stick to a title that general like 'example' or 'loading a C compiler'.